iothub/service/src/Amqp/AmqpReceivingLinkHandler.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Devices.Common.Exceptions;
+
+namespace Microsoft.Azure.Devices.Amqp
+{
+    /// <summary>
+    /// Handles a single AMQP receiving link.
+    /// </summary>
+    internal class AmqpReceivingLinkHandler
+    {
+        private ReceivingAmqpLink _receivingLink;
+        private Action<AmqpMessage> _messageHandler;
+
+        private EventHandler _connectionLossHandler;
+
+        private string _linkAddress;
+        private string _linkName;
+
+        public AmqpReceivingLinkHandler(string linkAddress, Action<AmqpMessage> messageHandler, EventHandler connectionLossHandler)
+        {
+            _linkAddress = linkAddress;
+            _messageHandler = messageHandler;
+            _connectionLossHandler = connectionLossHandler;
+        }
+
+        public async Task OpenAsync(AmqpSession session, CancellationToken cancellationToken)
+        {
+            _linkName = "ReceiverLink-" + Guid.NewGuid();
```

Comment:
Perhaps a code comment explaining why we uniquely name each link?

Reply:
Good idea